src/sentry/incidents/logic.py

```diff
@@ -654,6 +658,15 @@ def create_alert_rule(
             ]
             AlertRuleExcludedProjects.objects.bulk_create(exclusions)
 
+        if environment:
+            for e in environment:
+                try:
+                    env = Environment.objects.get(id=e)
```

Comment:
I just realized we don't do any access valiation on the environments. We should make sure the environment actually belongs to the organization. Probably makes sense to do it as part of the serializer, since that's where we usually handle access control. You can see how we usually handle this via `ProjectField`